src/shogun/base/SGObject.h

```diff
@@ -1011,5 +1018,54 @@ class CSGObject
 	/** Subscriber used to call onNext, onComplete etc.*/
 	SGSubscriber* m_subscriber_params;
 };
+
+template <typename T1, typename T2>
+bool execute_dispatch_array_type(const CSGObject* obj, const std::string& name, T2&& lambda)
+{
+	Tag<CDynamicObjectArray*> tag_array_sg(name);
+	if (obj->has(tag_array_sg))
+	{
+		auto dispatched = obj->get(tag_array_sg);
+		lambda(*dispatched); // is stored as a pointer
+		return true;
+	}
+
+	Tag<std::vector<T1*>> tag_vector(name);
+	if (obj->has(tag_vector))
+	{
+		auto dispatched = obj->get(tag_vector);
+		lambda(dispatched);
+		return true;
+	}
+
+	return false;
+}
+
+template <typename T>
+CSGObject* get_sgobject_type_dispatcher(const CSGObject* obj, const std::string& name)
+{
+	if (obj->has<T*>(name))
+	{
+		T* result = obj->get<T*>(name);
+		SG_REF(result)
+		return (CSGObject*)result;
+	}
+
+	return nullptr;
+}
+
+template <typename T>
```

Comment:
@gf712 I would like to unify those two functions, to accept a get lambda that has the index parameter bound. However, the `get` call inside that lambda is templated. Not sure that is possible?

Reply:
works now